kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,243 @@
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.applozic.mobicomkit.uiwidgets.R;
+import com.applozic.mobicommons.commons.core.utils.Utils;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * fragment for the feedback input form
+ * @author shubham
+ * @date july 19
+ */
+public class FeedbackInputFragment extends Fragment implements View.OnClickListener{
+
+    private static final String TAG = "FeedbackInputFragment";
+
+    EditText editTextFeedbackComment;
+    Button buttonSubmitFeedback;
+    Button buttonCloseFragment;
+
+    FeedbackRatingGroup feedbackRatingGroup;
+
+    FeedbackFragmentListener feedbackFragmentListener;
+
+    public static final int RATING_POOR = 1;
+    public static final int RATING_AVERAGE = 2;
+    public static final int RATING_GOOD = 3;
+
+    //using IntDef to replace enum
+    @Retention(RetentionPolicy.SOURCE)
+    @IntDef({RATING_POOR, RATING_AVERAGE, RATING_GOOD})
+    public @interface Rating {}
+
+    @Rating
+    public int getRatingLevel() {
+        return ratingLevel;
+    }
+
+    public void setRatingLevel(@Rating int ratingLevel) {
+        this.ratingLevel = ratingLevel;
+    }
+
+    @Rating private int ratingLevel;
+
+    public void setFeedbackFragmentListener(FeedbackFragmentListener feedbackFragmentListener) {
+        this.feedbackFragmentListener = feedbackFragmentListener;
+    }
+
+    public static String getTAG() {
+        return TAG;
+    }
+
+    public FeedbackInputFragment() {
+        super();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (!FeedbackFragmentListener.class.isAssignableFrom(MobiComConversationFragment.class)) {
+            Utils.printLog(context, TAG, "Implement FeedbackFragmentListener in your parent fragment.");
+            throw new ClassCastException("Implement FeedbackFragmentListener in your parent fragment.");
+        }
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.feedback_rating_layout, container, false);
+
+        editTextFeedbackComment = view.findViewById(R.id.idEditTextFeedback);
+        buttonSubmitFeedback = view.findViewById(R.id.idButtonSubmit);
+
+        buttonCloseFragment = view.findViewById(R.id.idCloseFeedbackFragment);
+
+        editTextFeedbackComment.setVisibility(View.GONE);
+
+        //selected
+        Drawable[] buttonDrawablesSelected = new Drawable[3];
```

Comment:
@shubhamtewari ic_sad_1 is related to ic_sad_1_grey and same is the case with others but looking at the 2 arrays, we are taking the array index and then mapping them everywhere. Ideally it should be self explanatory.

what if we combine these 2 into a 2-D array something like 
Drawable[] ratingDrawables = new Drawable[3][2];
ratingDrawables[0][0] = ... store ic_sad_1
ratingDrawables[0][1] = ... store ic_sad_1_grey

in short, 1-d stands for different ratings and second dimension stands for selected and unselected.

Then FeedbackRatingGroup will be simplied with only 1 property instead of 2 different properties and if else inside setDrawableForRating can be removed.